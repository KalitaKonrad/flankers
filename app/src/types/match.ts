export interface MatchElementInHistory {
  winner: boolean;
}

export interface MatchHistoryResponse {
  current_page: number;
  data: MatchElementInHistory[];
  first_page_url: string;
  from: number;
  last_page_url: string;
  links: {
    url: string;
    label: string;
    active: boolean;
  };
  next_page_url: string;
  path: string;
  per_page: number;
  prev_page_url: string;
  to: number;
  total: number;
}

export enum MatchJoinType {
  TEAM = 'team',
  OPEN = 'ffa',
}

export enum MatchVisibility {
  PUBLIC,
  PRIVATE,
}

export interface MatchCreatePayload {
  type: string;
  isRated: boolean;
  isPublic: boolean;
  bet: number;
  playersAmount: number;
  lat: number;
  long: number;
}
