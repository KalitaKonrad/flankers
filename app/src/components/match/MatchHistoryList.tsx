import React from 'react';
import { FlatList, ListRenderItemInfo, StyleSheet } from 'react-native';
import { List, Text } from 'react-native-paper';

import { MatchElementInHistory } from '../../types/match';
import { formatDate } from '../../utils/formatBackendTIme';

interface MatchHistoryListProps {
  matchHistory: MatchElementInHistory[];
  onListEndReached(): void;
}

export const MatchHistoryList: React.FC<MatchHistoryListProps> = (props) => {
  const renderItem = ({ item }: ListRenderItemInfo<MatchElementInHistory>) => (
    <List.Item
      title={formatDate(item.updated_at)}
      titleStyle={styles.elementName}
      description={
        (item.winner ? 'Zwycięstwo' : 'Porażka') +
        ' w trybie ' +
        (item.type === 'team' ? 'drużynowym' : 'ogólnym')
      }
      left={(props) => (
        <List.Icon {...props} icon={item.winner ? 'trophy' : 'trophy-broken'} />
      )}
    />
  );
  return (
    <FlatList
      data={props.matchHistory}
      renderItem={renderItem}
      contentContainerStyle={styles.container}
      keyExtractor={(match) => match.id.toString()}
      onEndReached={props.onListEndReached}
      onEndReachedThreshold={0.7}
    />
  );
};

const styles = StyleSheet.create({
  container: {
    paddingRight: 16,
  },
  elementName: {
    fontWeight: 'bold',
  },
  rightIndexStyle: {
    fontWeight: 'bold',
    textAlign: 'justify',
    marginRight: 12,
    paddingTop: 6,
  },
});
