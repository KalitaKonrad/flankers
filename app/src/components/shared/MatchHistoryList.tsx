import React from 'react';
import { List } from 'react-native-paper';

import { Match } from '../../types/match';

interface MatchHistoryListProps {
  matchHistory: Match[];
}

export const MatchHistoryList: React.FC<MatchHistoryListProps> = ({
  matchHistory,
}) => {
  return (
    <List.Item
      title="Match"
      description="Result"
      left={(props) => <List.Icon {...props} icon="star" />}
    />
  );
};
