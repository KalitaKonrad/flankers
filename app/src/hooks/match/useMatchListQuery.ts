import { useQuery } from 'react-query';

import { QUERY_GAMES } from '../../const/query.const';
import { MatchResponse } from '../../types/matchResponse';
import { useAxios } from '../useAxios';

export const useMatchListQuery = () => {
  const axios = useAxios();

  return useQuery<MatchResponse[]>(QUERY_GAMES, async () => {
    const response = await axios.get<MatchResponse[]>('games');

    return response.data;
  });
};
