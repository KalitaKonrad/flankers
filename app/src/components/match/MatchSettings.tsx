import * as React from 'react';
import { useState } from 'react';
import { StyleSheet, Text, View } from 'react-native';

import { Switch } from '../shared/Switch';

interface MatchSettingsProps {
  onRankingMatchToggled: (arg: boolean) => void;
  onVisibilityToggled: (arg: boolean) => void;
  onMatchTypeToggled: (arg: boolean) => void;
}

export const MatchSettings: React.FC<MatchSettingsProps> = (props) => {
  const [isSwitchedMatchType, setIsSwitchedMatchType] = useState<boolean>(
    false
  );
  const [isSwitchedRankingMatch, setIsSwitchedRankingMatch] = useState<boolean>(
    false
  );
  const [
    isSwitchedMatchVisibility,
    setIsSwitchedMatchVisibility,
  ] = useState<boolean>(false);

  return (
    <View style={styles.matchSettings}>
      <Text>Typ Meczu</Text>
      <Switch
        leftLabel="Drużynowy"
        rightLabel="Swobodny"
        onToggle={(res) => {
          setIsSwitchedMatchType(res);
          props.onMatchTypeToggled(isSwitchedMatchType);
        }}
      />
      <Text>Mecz Rankingowy</Text>
      <Switch
        leftLabel="Tak"
        rightLabel="Nie"
        onToggle={(res) => {
          setIsSwitchedRankingMatch(res);
          props.onRankingMatchToggled(isSwitchedRankingMatch);
        }}
      />
      <Text>Widoczność</Text>
      <Switch
        leftLabel="Publiczny"
        rightLabel="Prywatny"
        onToggle={(res) => {
          setIsSwitchedMatchVisibility(res);
          props.onVisibilityToggled(isSwitchedMatchVisibility);
        }}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  matchSettings: {
    flex: 0.7,
    justifyContent: 'space-between',
    alignItems: 'center',
  },
});
