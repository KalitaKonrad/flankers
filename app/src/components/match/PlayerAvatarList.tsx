import React from 'react';
import { StyleSheet, View } from 'react-native';

import { MembersPayload } from '../../types/squadResponse';
import { UserProfilePayload } from '../../types/userProfilePayload';
import { Avatar } from '../shared/Avatar';

interface PlayerAvatarListProps {
  players: MembersPayload[];
}

export const PlayerAvatarList: React.FC<PlayerAvatarListProps> = ({
  players,
}) => {
  return (
    <View style={styles.container}>
      {players.map((player, i) => (
        <Avatar
          border={4}
          elevation={4}
          key={player.id}
          containerStyle={[styles.avatar, { zIndex: players.length - i }]}
          size={65}
          src={{ uri: player.versioned_avatar }}
        />
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingLeft: 20,
  },
  avatar: {
    marginLeft: -20,
  },
});
