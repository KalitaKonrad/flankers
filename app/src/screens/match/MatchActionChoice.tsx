import { StackNavigationProp, StackScreenProps } from '@react-navigation/stack';
import React from 'react';
import { Text, View } from 'react-native';
import { Button } from 'react-native-paper';

import { SubmitButton } from '../../components/shared/SubmitButton';
import { theme } from '../../theme';
import { MatchScreenStackParamList } from './MatchScreenStack';

type MatchActionChoiceProps = object &
  StackScreenProps<MatchScreenStackParamList, 'MatchCreate'>;

export const MatchActionChoice: React.FC<MatchActionChoiceProps> = ({
  navigation,
}) => {
  return (
    <>
      <SubmitButton
        backgroundColor={theme.colors.primary}
        labelColor={theme.colors.background.white}
        onPress={() => navigation.push('MatchJoinFromMap')}>
        Mecze w okolicy
      </SubmitButton>
      <SubmitButton
        backgroundColor={theme.colors.primary}
        labelColor={theme.colors.background.white}
        onPress={() => navigation.push('MatchCreate')}>
        Utw√≥rz mecz
      </SubmitButton>
    </>
  );
};
