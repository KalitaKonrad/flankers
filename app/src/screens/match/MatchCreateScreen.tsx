import { StackScreenProps } from '@react-navigation/stack';
import React, { useState } from 'react';
import {
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  View,
} from 'react-native';

import { MatchSettings } from '../../components/MatchSettings';
import { SubmitButton } from '../../components/shared/SubmitButton';
import { useTeamCreateMutation } from '../../hooks/useTeamCreateMutation';
import { TextStyle, theme } from '../../theme';
import { setResponseErrors } from '../../utils/setResponseErrors';
import { MatchScreenStackParamList } from './MatchScreenStack';

type MatchCreateScreenProps = object &
  StackScreenProps<MatchScreenStackParamList, 'MatchCreate'>;

export const MatchCreateScreen: React.FC<MatchCreateScreenProps> = ({
  navigation,
}) => {
  const [feeAmount, setFeeAmount] = useState<number>(5.0);
  const [playersAmount, setPlayersAmount] = useState<number>(2.0);
  const [isRankingMatchSwitched, setIsRankingMatchSwitched] = useState<boolean>(
    true
  );
  const [isVisibilitySwitched, setIsVisibilitySwitched] = useState<boolean>(
    true
  );
  const [isTypeSwitched, setIsTypeSwitched] = useState<boolean>(true);

  const [mutate] = useTeamCreateMutation();

  const onPress = () => {
    mutate({
      typeTeam: isTypeSwitched,
      rated: isRankingMatchSwitched,
      public: isVisibilitySwitched,
      bet: feeAmount,
    });
  };

  return (
    // //////////////////////////////////////////////////////////////////////
    // TODO: INLINE HEADER
    // ///////////////////////////////////////////////////
    <View style={styles.container}>
      <View style={styles.matchSettings}>
        <MatchSettings
          onRankingMatchToggled={(res) => {
            setIsRankingMatchSwitched(res);
          }}
          onMatchTypeToggled={(res) => setIsTypeSwitched(res)}
          onVisibilityToggled={(res) => setIsVisibilitySwitched(res)}
        />
      </View>
      <View>
        <View style={styles.label}>
          <Text style={{ textAlign: 'center' }}>Liczba Graczy</Text>
        </View>

        <View style={styles.playersAmountPicker}>
          <TouchableOpacity
            style={styles.button}
            onPress={() => {
              if (playersAmount > 2) {
                setPlayersAmount(playersAmount - 1);
              }
            }}>
            <Text style={{ color: theme.colors.black }}>-</Text>
          </TouchableOpacity>
          <Text style={TextStyle.noteH2}>{playersAmount}</Text>
          <TouchableOpacity
            style={styles.button}
            onPress={() => {
              if (playersAmount < 5) {
                setPlayersAmount(playersAmount + 1);
              }
            }}>
            <Text style={{ color: theme.colors.black }}>+</Text>
          </TouchableOpacity>
        </View>
      </View>

      {isRankingMatchSwitched ? (
        <View>
          <View style={styles.label}>
            <Text style={{ textAlign: 'center' }}>Wpisowe</Text>
          </View>
          <View style={styles.feePicker}>
            <TouchableOpacity
              style={styles.button}
              onPress={() => {
                if (feeAmount > 0) {
                  setFeeAmount(feeAmount - 1);
                }
              }}>
              <Text style={{ color: theme.colors.black }}>-</Text>
            </TouchableOpacity>
            <Text style={TextStyle.noteH2}>{feeAmount + '.00'}</Text>
            <TouchableOpacity
              style={styles.button}
              onPress={() => setFeeAmount(feeAmount + 1)}>
              <Text style={{ color: theme.colors.black }}>+</Text>
            </TouchableOpacity>
          </View>
        </View>
      ) : (
        <></>
      )}

      {!isVisibilitySwitched && !isTypeSwitched ? (
        <View style={styles.invitation}>
          <Text style={{ textAlign: 'center' }}>
            Kod gry dzięki któremu przeciwna drużyna będzie mogła dołączyć
            zostanie wygenerowany. Wybierz miejsce na mapie
          </Text>
        </View>
      ) : (
        <></>
      )}

      {!isVisibilitySwitched && isTypeSwitched ? (
        <View style={styles.invitation}>
          <Text style={{ textAlign: 'center' }}>
            Kod gry dzięki któremu inne osoby będą mogły dołączyć zostanie
            wygenerowany. Wybierz miejsce na mapie
          </Text>
        </View>
      ) : (
        <></>
      )}

      <View style={styles.submitBtn}>
        <SubmitButton
          backgroundColor={theme.colors.primary}
          labelColor={theme.colors.white}
          onPress={() => navigation.push('MatchLocation')}>
          utwórz
        </SubmitButton>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flex: 1,
  },
  submitBtn: {
    bottom: 20,
  },
  matchSettings: {
    flex: 1,
    justifyContent: 'space-between',
    top: 20,
  },
  playersAmountPicker: {
    position: 'relative',
    display: 'flex',
    flexDirection: 'row',
    paddingHorizontal: 10,
    paddingTop: 10,
    paddingBottom: 10,
    justifyContent: 'space-around',
    alignItems: 'center',
    minHeight: 50,
    marginBottom: 20,
  },

  feePicker: {
    position: 'relative',
    display: 'flex',
    flexDirection: 'row',
    paddingLeft: 20,
    paddingRight: 20,
    paddingTop: 10,
    paddingBottom: 10,
    justifyContent: 'space-around',
    alignItems: 'center',
    minHeight: 50,
    marginBottom: 25,
  },
  button: {
    width: 60,
    height: 60,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 10,
    borderRadius: 100,
    backgroundColor: theme.colors.darkGray,
  },
  label: {
    top: 10,
  },
  textInputStyle: {
    borderTopLeftRadius: 12,
    borderTopRightRadius: 12,
    borderBottomLeftRadius: 12,
    borderBottomRightRadius: 12,
    margin: 10,
    paddingVertical: 6,
    paddingHorizontal: 10,
    borderColor: 'gray',
    borderWidth: 1,
    backgroundColor: theme.colors.darkGray,
  },
  invitation: {
    bottom: 15,
    margin: 10,
  },
});
