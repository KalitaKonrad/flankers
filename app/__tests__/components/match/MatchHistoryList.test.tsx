import { render } from '@testing-library/react-native';
import React from 'react';

import { MatchHistoryList } from '../../../src/components/match/MatchHistoryList';

describe('<MatchHistoryList>', () => {
  it('should render correctly', () => {
    const tree = render(
      <MatchHistoryList onListEndReached={() => null} matchHistory={[]} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
