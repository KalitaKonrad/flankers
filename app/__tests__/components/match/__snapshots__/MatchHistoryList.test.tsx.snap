// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MatchHistoryList> renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingRight": 16,
    }
  }
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onEndReachedThreshold={0.7}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View />
</RCTScrollView>
`;

exports[`<MatchHistoryList> renders correctly when empty array with data provided 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingRight": 16,
    }
  }
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onEndReachedThreshold={0.7}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View />
</RCTScrollView>
`;

exports[`<MatchHistoryList> renders correctly when valid data provided 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingRight": 16,
    }
  }
  data={
    Array [
      Object {
        "id": 5,
        "type": "hehe",
        "updated_at": "21:37",
        "winner": true,
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onEndReachedThreshold={0.7}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            Array [
              Object {
                "padding": 8,
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 8,
                  "width": 40,
                },
                Object {
                  "marginLeft": 0,
                  "marginRight": 16,
                },
              ]
            }
          >
            <Text />
          </View>
          <View
            style={
              Array [
                Object {
                  "marginVertical": 6,
                  "paddingLeft": 8,
                },
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "fontSize": 16,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                  ],
                ]
              }
            >
              21:37 undefined
            </Text>
            <Text
              numberOfLines={2}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Zwycięstwo w trybie ogólnym
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
